### PR TITLE
Fix docs typo

### PR DESCRIPTION
**Pull request by mski-iksm**
I've fixed a small typo and added section number to make document easy to read.

Please review! @Hi-king @hirosassa @vaaaaanquish

### CODE REVIEW

**Review by vaaaaanquish — CHANGES_REQUESTED** [review]
Have you really checked the results? Please fix format breaking changes.

```
/github/gokart/tmp/gokart/docs/task_parameters.rst:29: WARNING: Title underline too short.

1. gokart.TaskInstanceParameter
----------------------------
/github/gokart/tmp/gokart/docs/task_parameters.rst:59: WARNING: Title underline too short.

2. gokart.ListTaskInstanceParameter
----------------------------
/github/gokart/tmp/gokart/docs/task_parameters.rst:59: WARNING: Title underline too short.

2. gokart.ListTaskInstanceParameter
----------------------------
/github/gokart/tmp/gokart/docs/task_parameters.rst:65: WARNING: Title underline too short.

3. gokart.ExplicitBoolParameter
----------------------------
/github/gokart/tmp/gokart/docs/task_parameters.rst:65: WARNING: Title underline too short.

3. gokart.ExplicitBoolParameter
----------------------------
/github/gokart/tmp/gokart/docs/task_parameters.rst:82: WARNING: Title underline too short.

Setting Task Parameters
================
/github/gokart/tmp/gokart/docs/task_parameters.rst:82: WARNING: Title underline too short.

Setting Task Parameters
================
/github/gokart/tmp/gokart/docs/task_parameters.rst:93: WARNING: Title underline too short.

1. Set parameter from command line
-------------------------
/github/gokart/tmp/gokart/docs/task_parameters.rst:102: WARNING: Title underline too short.

2. Set parameter at config file
-------------------------
/github/gokart/tmp/gokart/docs/task_parameters.rst:102: WARNING: Title underline too short.

2. Set parameter at config file
-------------------------
/github/gokart/tmp/gokart/docs/task_parameters.rst:131: WARNING: Title underline too short.

3. Set parameter at upstream task
-------------------------
/github/gokart/tmp/gokart/docs/task_parameters.rst:131: WARNING: Title underline too short.

3. Set parameter at upstream task
-------------------------
/github/gokart/tmp/gokart/docs/task_parameters.rst:143: WARNING: Title underline too short.

4. Inherit parameter from other task
-------------------------
/github/gokart/tmp/gokart/docs/task_parameters.rst:143: WARNING: Title underline too short.

4. Inherit parameter from other task
-------------------------
```

**Comment by vaaaaanquish** [comment]
By not following the formatting and numbering, tree is not shown.

スクリーンショット 2021-04-25 16 57 07

Use the following command
```sh
sphinx-autobuild docs/ build
```

File: docs/task_parameters.rst [hunk 1]
```diff
@@ -1,8 +1,9 @@
+===============
 Task Parameters
 ===============
 
 We can set parameters for tasks.
-Aslo please refer to :doc:`task_settings` section.
+Also please refer to :doc:`task_settings` section.
```
Comment (hirosassa) [inline docs/task_parameters.rst:6]: Thanks 👍

**Comment by vaaaaanquish** [comment]
I don't know why I need a numbering. Each item is independent.

**Comment by mski-iksm** [comment]
> By not following the formatting and numbering, tree is not shown.
> 
> スクリーンショット 2021-04-25 16 57 07
> 
> Use the following command
> 
> ```shell
> sphinx-autobuild docs/ build
> ```

**Review by hirosassa — COMMENTED** [review]
@mski-iksm Thanks! Plz check the formatting errors @vaaaaanquish commented

**Comment by mski-iksm** [comment]
> By not following the formatting and numbering, tree is not shown.
> 
> スクリーンショット 2021-04-25 16 57 07
> 
> Use the following command
> 
> ```shell
> sphinx-autobuild docs/ build
> ```

@vaaaaanquish 
Do you mean that subcategories (i.e. 1. gokart.TaskInstanceParameter) are not shown in this list?
I think thats how this listing works, since other subsections are not shown too.

**Comment by mski-iksm** [comment]
> I don't know why it need a numbering. Each item is independent. Excessive numbering can mislead users.

@vaaaaanquish 
I thought the difference between section and subsection is hard to distinguish which makes hard to read.

But as mentioned, numbering can mislead users that they have follow the orders, so I though A, B, C... may be better.
How do you think of it?

**Comment by vaaaaanquish** [comment]
@mski-iksm 
> I thought the difference between section and subsection is hard to distinguish which makes hard to read.

This is important misunderstanding. It's not a "writing section problem", but a "sphinx design problem".

It's same you saying following

bad
```
# Title
 `~~~~~~`
## foo
 `~~~~~~`
## bar
 `~~~~~~`
```

good
```
# Title
 `~~~~~~`
### 1, foo
 `~~~~~~`
### 2, bar
 `~~~~~~`
```

File: docs/task_parameters.rst (hunk 1 above)
Comment (vaaaaanquish) [reply docs/task_parameters.rst:6]: good works:+1:

**Comment by vaaaaanquish** [comment]
> numbering

If you have examples in other OSS, please share them:)

**Comment by mski-iksm** [comment]
@vaaaaanquish 
I've separated the descriptions of Setting Task Parameters to a new file `setting_task_parameters.rst`.

File: docs/setting_task_parameters.rst [hunk 2]
```diff
@@ -0,0 +1,98 @@
+============================
```
Comment (vaaaaanquish) [inline docs/setting_task_parameters.rst:1]: [ASK]Is this Line necessary?

**Comment by vaaaaanquish** [comment]
LGTM. Thanks good PR.

**Review by hirosassa — APPROVED** [review]
LGTM!

File: docs/setting_task_parameters.rst (hunk 2 above)
Comment (mski-iksm) [reply docs/setting_task_parameters.rst:1]: @vaaaaanquish 
I think we should have this line to indicate the document title.
https://docutils.sourceforge.io/docs/user/rst/quickstart.html#document-title-subtitle

Comment (vaaaaanquish) [reply docs/setting_task_parameters.rst:1]: Oh, I didn't know this. Thx!

**Review by vaaaaanquish — APPROVED** [review]

**Comment by hirosassa** [comment]
@mski-iksm Thank you for your improvement!

**Pull request merged**